### PR TITLE
added list branches with hash

### PR DESCRIPTION
**Pull request by ithink20**
improved cmdline arguments

### CODE REVIEW

File: hello.go [hunk 1]
```diff
@@ -161,15 +162,36 @@ func printObjectFileContent(contentReader io.Reader) {
...
-	objectFile, err := os.Open(path)
-	if err != nil {
-		log.Fatal(err)
-	}
-	contentReader, err := zlib.NewReader(objectFile)
-	if err != nil {
-		log.Fatal(err)
+	var path string
+	args := os.Args[1]
+	if args == "branches" { 	// git branch -l
```
Comment (anoopc1) [inline hello.go]: We should move logic for `branches` and `cat-file` to dedicated functions

File: hello.go [hunk 2]
```diff
@@ -161,15 +162,36 @@ func printObjectFileContent(contentReader io.Reader) {
...
+			log.Fatal(err)
+		}
+		for _, branch := range branches {
+			file, err := os.Open(path + "/" + branch.Name())
+			if err != nil {
+				log.Fatal(err)
+			}
+			bufScanner := bufio.NewScanner(file)
+			bufScanner.Split(bufio.ScanBytes) // read byte by byte
+			fileMetadataBytes := scanBytesUntilDelimiter(bufScanner, 0, false) // '\0' character in ascii is same as 0
```
Comment (anoopc1) [inline hello.go]: what is the format of this file?

there is no need to ready by byte till `\0`. I don't think there is a `\0` at the end of the file.

```
$> od -xc .git/refs/heads/master
0000000      3231    3761    3730    6565    6333    6662    3261    3039
           1   2   a   7   0   7   e   e   3   c   b   f   a   2   9   0
0000020      6665    6561    6139    6563    3136    3839    3231    3630
           e   f   a   e   9   a   c   e   6   1   9   8   1   2   0   6
0000040      6636    6166    6436    6364    000a
           6   f   f   a   6   d   d   c  \n
0000051
```

File: hello.go [hunk 3]
```diff
@@ -161,15 +162,36 @@ func printObjectFileContent(contentReader io.Reader) {
...
+			fileMetadataString := string(fileMetadataBytes[:len(fileMetadataBytes)-1])
```
Comment (anoopc1) [inline hello.go]: I don't think it will print the correct hash. It will leave last character.

**Review by anoopc1 — CHANGES_REQUESTED** [review]

File: hello.go (hunk 2 above)
Comment (ithink20) [reply hello.go]: Yeah! changed delimiter to 0x0A for \n.

**Review by anoopc1 — APPROVED** [review]
lgtm

**Pull request merged**